cm/refurbish/refurbish.py

```diff
@@ -0,0 +1,643 @@
+import logging as log
+import os
+from functools import lru_cache
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+import geopandas as gpd
+import pandas as pd
+from BaseCM import cm_hddcdd
+from BaseCM.cm_output import validate
+from shapely import geometry
+
+logging = log.getLogger("cm-refurbish")
+logging.setLevel(log.DEBUG)
+
+CURRENT_FILE_DIR = Path(__file__).parent
+
+BS2TABULA = {
+    "Appartment blocks": ("AB",),
+    "Education": (),
+    "Health": (),
+    "Hotels and Restaurants": (),
+    "Multifamily houses": ("MFH",),
+    "Offices": (),
+    "Other non-residential buildings": (),
+    "Single family - Terraced houses": ("SFH", "TH"),
+    "Total": (),
+    "Trade": (),
+}
+
+BS2YEARS = {
+    "Before 1945": (pd.NaT, 1945),
+    "1945 - 1969": (1945, 1969),
+    "1970 - 1979": (1970, 1979),
+    "1980 - 1989": (1980, 1989),
+    "1990 - 1999": (1990, 1999),
+    "2000 - 2010": (2000, 2010),
+    "Post 2010": (2010, pd.NaT),
+    "Berfore 1945": (pd.NaT, 1945),
+}
+
+
+def path_building_stock() -> Path:
+    """Return the path to the building stock dataset."""
+    return cm_hddcdd.get_data_dir() / os.environ.get("BUILSTK", "building_stock.csv")
+
+
+@lru_cache()
+def get_building_stock() -> pd.DataFrame:
+    """Return building_stock dataframe"""
+    logging.info(f"Reading building_stock data from: {path_building_stock()}")
+    bs = pd.read_csv(path_building_stock(), sep="|", index_col=0)
+    # fix label adding a space
+    bs.loc[
+        bs["btype"] == "Single family- Terraced houses", "btype"
+    ] = "Single family - Terraced houses"
+
+    starts = []
+    ends = []
+    for _, agekey in bs["bage"].items():
+        s, e = (pd.NaT, pd.NaT) if pd.isnull(agekey) else BS2YEARS[agekey]
+        starts.append(s)
+        ends.append(e)
+    bs["start"] = starts
+    bs["end"] = ends
+    return bs
+
+
+def path_population() -> Path:
+    """Return the path to the population dataset."""
+    return cm_hddcdd.get_data_dir() / os.environ.get(
+        "POPGJSN", "LAU_RG_01M_2020_4326.geojson"
+    )
+
```

Comment:
When I test your CM, I get the following error: 
```
/cm_inputs/hdd-cdd/LAU_RG_01M_2020_4326.geojson: No such file or directory
```
I think the error comes from the following function.
In the directory `/cm_inputs/hdd-cdd/`, there is only the directory `hdd-cdd-main`